src/feature/matching.cc

```diff
@@ -1467,8 +1468,16 @@ void ImagePairsFeatureMatcher::Run() {
       continue;
     }
 
-    image_pairs.emplace_back(image_name_to_image_id.at(image_name1),
-                             image_name_to_image_id.at(image_name2));
+    image_pairs_set.insert(
```

Comment:
Instead of forward/backward converting and doing the loop twice, you could simply check whether image_pairs_set already contains the pair and then directly emplace_back to image_pairs